src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1196,6 +1196,45 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_RemoveAllSrpSer
     return error;
 }
 
+template <class ImplClass>
+CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_ClearSrpService(const char * aInstanceName, const char * aName)
+{
+    CHIP_ERROR error                         = CHIP_NO_ERROR;
+    typename SrpClient::Service * srpService = nullptr;
+
+    Impl()->LockThreadStack();
+
+    VerifyOrExit(aInstanceName, error = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(strlen(aInstanceName) <= SrpClient::kMaxInstanceNameSize, error = CHIP_ERROR_INVALID_STRING_LENGTH);
+    VerifyOrExit(aName, error = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(strlen(aName) <= SrpClient::kMaxNameSize, error = CHIP_ERROR_INVALID_STRING_LENGTH);
+
+    // Check if service to clear exists.
+    for (typename SrpClient::Service & service : mSrpClient.mServices)
+    {
+        if ((strcmp(service.mInstanceName, aInstanceName) == 0) && (strcmp(service.mName, aName) == 0))
+        {
+            srpService = &service;
+            break;
+        }
+    }
+
+    VerifyOrExit(srpService, error = MapOpenThreadError(OT_ERROR_NOT_FOUND));
+
+    ChipLogProgress(DeviceLayer, "clearing srp service: %s.%s", aInstanceName, aName);
+    error = MapOpenThreadError(otSrpClientClearService(mOTInst, &(srpService->mService)));
+
+    VerifyOrExit(error == CHIP_NO_ERROR, );
+
+    // Free memory immediately, as OnSrpClientNotification will not be called.
+    memset(srpService, 0, sizeof(typename SrpClient::Service));
```

Comment:
nit: just a preference to use: `sizeof(srpService)`

Reply:
Ok, `sizeof(*srpService)` then :smile: